File: android/src/main/kotlin/com/morbit/amap_flutter/AMapSdkApi.kt
package com.morbit.amap_flutter

import android.content.Context
import com.amap.api.maps.MapsInitializer
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class AMapSdkApi {
  companion object {
    fun setup(binding: FlutterPluginBinding) {
      val initializerChannel = MethodChannel(binding.binaryMessenger, "plugins.flutter.dev/amap_initializer")
      initializerChannel.setMethodCallHandler { call: MethodCall, result: MethodChannel.Result ->
        if (call.method == "agreePrivacy") {
          val agree = call.argument<Boolean>("agree")!!
          agreePrivacy(binding.applicationContext, agree)
          result.success(null)
        } else if (call.method == "setApiKey") {
          val apiKey = call.argument<String>("androidKey")!!
          setApiKey(apiKey)
          result.success(null)
        }
      }
    }

    private fun agreePrivacy(context: Context, agreePrivacy: Boolean) {
      MapsInitializer.updatePrivacyShow(context, agreePrivacy, agreePrivacy)
      MapsInitializer.updatePrivacyAgree(context, agreePrivacy)
    }

    private fun setApiKey(apiKey: String) {
      MapsInitializer.setApiKey(apiKey)
    }
  }
}

File: ios/Classes/AMapSdkApi.swift
import Flutter
import MAMapKit

class _AMapSdkApi: NSObject {
  static func setup(registrar: FlutterPluginRegistrar) {
    let initializerChannel = FlutterMethodChannel(name: "plugins.flutter.dev/amap_initializer", binaryMessenger: registrar.messenger())
    initializerChannel.setMethodCallHandler({ (call: FlutterMethodCall, result: FlutterResult) -> Void in
      if(call.method == "agreePrivacy") {
        let arguments = call.arguments as! Dictionary<String, AnyObject>
        let agree = arguments["agree"] as! Bool
        _AMapSdkApi.agreePrivacy(agreePrivacy: agree)
        result(nil)
      } else if(call.method == "setApiKey") {
        let arguments = call.arguments as! Dictionary<String, AnyObject>
        let apiKey = arguments["iosKey"] as! String
        _AMapSdkApi.setApiKey(apiKey: apiKey)
        result(nil)
      }
    })
  }

  static func agreePrivacy(agreePrivacy: Bool) {
    MAMapView.updatePrivacyShow(AMapPrivacyShowStatus.init(agreePrivacy), privacyInfo: AMapPrivacyInfoStatus.init(agreePrivacy))
    MAMapView.updatePrivacyAgree(AMapPrivacyAgreeStatus.init(agreePrivacy))
  }

  static func setApiKey(apiKey: String) {
    AMapServices.shared().apiKey = apiKey
  }
}

